File: tests/benchmark/corpus/typescript/safe/safe_direct_path_sanitizer.ts
// ts-safe-014: direct-return path sanitiser.
import * as fs from 'fs';

function sanitizePath(s: string): string {
  if (s.includes('..') || s.startsWith('/') || s.startsWith('\\')) {
    return '';
  }
  return s;
}

export default function (req: { query: { path: string } }) {
  const raw = req.query.path;
  const safe = sanitizePath(raw);
  return fs.readFileSync(safe);
}

File: tests/benchmark/corpus/python/safe/safe_direct_path_sanitizer.py
# py-safe-014: direct-return path sanitiser (Optional[str]) closing the
# OR-chain rejection path.  Mirrors rs-safe-014.
import os
from flask import request


def sanitize_path(s: str) -> str:
    if ".." in s or s.startswith("/") or s.startswith("\\"):
        return ""
    return s


def main() -> None:
    raw = request.args.get("path")
    safe = sanitize_path(raw)
    with open(safe) as f:
        f.read()
